components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -1040,50 +1040,87 @@ export class TypeORMWorkspaceDBImpl extends TransactionalDBImpl<WorkspaceDB> imp
     /**
      * Finds prebuilt workspaces by organization with optional filtering and pagination.
      * @param organizationId The ID of the organization.
-     * @param offset Offset for pagination.
-     * @param limit Limit for pagination.
+     * @param pagination Pagination per page size and result offset.
      * @param filter Filters for the search.
+     * @param sort Sort field and direction
      * @returns A promise that resolves to an array of PrebuiltWorkspace objects.
      */
     async findPrebuiltWorkspacesByOrganization(
         organizationId: string,
-        offset = 0,
-        limit = 25,
-        filter?: {
+        pagination: {
+            offset: number;
+            limit: number;
+        },
+        filter: {
             configuration?: {
                 id: string;
                 branch?: string;
             };
-            status: PrebuiltWorkspaceState;
+            state?: "succeeded" | "failed" | "unfinished";
             searchTerm?: string;
         },
+        sort: {
+            field: string;
+            order: "ASC" | "DESC";
+        },
     ): Promise<PrebuiltWorkspace[]> {
         const repo = await this.getPrebuiltWorkspaceRepo();
         const query = repo
             .createQueryBuilder("pws")
-            .orderBy("pws.creationTime", "DESC")
             .innerJoinAndMapOne(
                 "pws.workspace",
                 DBWorkspace,
                 "ws",
                 "pws.buildWorkspaceId = ws.id AND ws.organizationId = :organizationId",
                 { organizationId },
             )
-            .skip(offset)
-            .take(limit);
-
-        if (filter?.status) {
-            query.andWhere("pws.state = :state", { state: filter.status });
+            .skip(pagination.offset)
+            .take(pagination.limit)
+            .orderBy("pws.creationTime", sort.order); // todo: take sort field into account
+
+        if (filter.state) {
+            const { state } = filter;
+            // translating API state to DB state
```

Comment:
Puh, I'd reaaaally love this to happen on an(y) upper layer. :thinking: 
But I see the problem with the `available` state, and it's ambiguity. :+1: with this because it's an old code base.

@filiptronicek Do we have tests for this behavior somewhere? If not, would be great to have it in a follow-up! :pray:

Reply:
Agree tests would be great to have here, added them in https://github.com/gitpod-io/gitpod/pull/19354/commits/6a83afcde25096ce045f3b9b2c10dae8981b4787.